process/block/baseProcess.go

```diff
@@ -801,14 +801,15 @@ func (bp *baseProcessor) checkHeaderBodyCorrelation(miniBlockHeaders []data.Mini
 }
 
 func checkConstructionStateAndIndexesCorrectness(mbh data.MiniBlockHeaderHandler) error {
-	if mbh.GetConstructionState() == int32(block.PartialExecuted) && mbh.GetIndexOfLastTxProcessed() < int32(mbh.GetTxCount())-1 {
-		return nil
+	if mbh.GetConstructionState() == int32(block.PartialExecuted) && mbh.GetIndexOfLastTxProcessed() == int32(mbh.GetTxCount())-1 {
```

Comment:
these work if the GetIndexOfLastTxProcessed returns values in the [0, mbh.GetTxCount()) range

Reply:
Only this range is valid when a mini block is proposed and validated